### PR TITLE
Added CSR Access Test 

### PR DESCRIPTION
**Pull request by emgens**
Added riscv_csr test.

### CODE REVIEW

**Comment by emgens** [comment]
This is the csr access test generated by the gen_csr_test.py in the cve2 repository.

**Comment by emgens** [comment]
Per discussion on https://github.com/openhwgroup/cve2/issues/159 I've added gen_csr_test.py and the lib.py necessary to run it to this pull request.

File: cv32e20/tests/programs/custom/riscv_csr/gen_csr_test.py [hunk 1]
```diff
@@ -0,0 +1,445 @@
+"""
```
Comment (MikeOpenHWGroup) [inline cv32e20/tests/programs/custom/riscv_csr/gen_csr_test.py:1]: I won't gate the PR approval on this, but in future, anytime you make additions or modifications to an existing open-source file (with a permissive license), it is considered good practise to declare who owns the copyright on the modifications in the header.  A one liner comment on the top with your copyright declaration is sufficient.

**Review by MikeOpenHWGroup — APPROVED** [review]

**Comment by MikeOpenHWGroup** [comment]
Approved.  Will defer to @LeeHoff and/or @joecircello for final approvals.

**Pull request merged**